pkg/useragent/analyzer.go

```diff
@@ -110,38 +114,36 @@ func (a *analyzer) start() {
 					bson.M{"$match": bson.M{"count": bson.M{"$lte": maxLeft}}},
 				}
 
-				var resList struct {
-					ID    string   `bson:"_id"`
-					IPS   []string `bson:"ips"`
-					Count int      `bson:"count"`
+				var rareSigList struct {
+					OrigIps []data.UniqueIP `bson:"ips"`
 				}
 
-				_ = ssn.DB(a.db.GetSelectedDB()).C(a.conf.T.UserAgent.UserAgentTable).Pipe(query).AllowDiskUse().One(&resList)
+				_ = ssn.DB(a.db.GetSelectedDB()).C(a.conf.T.UserAgent.UserAgentTable).Pipe(query).AllowDiskUse().One(&rareSigList)
 
-				for _, entry := range resList.IPS {
+				for _, rareSigIP := range rareSigList.OrigIps {
 
-					newRecordFlag := false
+					newRecordFlag := false // have we created a rare signature entry for this host in this chunk yet?
 
-					type hostRes struct {
+					type hostEntry struct {
 						CID int `bson:"cid"`
 					}
 
-					var res2 []hostRes
+					var hostEntries []hostEntry
 
-					_ = ssn.DB(a.db.GetSelectedDB()).C(a.conf.T.Structure.HostTable).Find(bson.M{"ip": entry, "dat.rsig": data.name}).All(&res2)
+					entryHostQuery := rareSigIP.BSONKey()
+					entryHostQuery["dat.rsig"] = datum.Name
+					//TODO: Consider adding the chunk to the query instead of checking it after the query
 
-					if !(len(res2) > 0) {
-						newRecordFlag = true
-						// fmt.Println("host no results", res2, data.Host)
-					} else {
+					_ = ssn.DB(a.db.GetSelectedDB()).C(a.conf.T.Structure.HostTable).Find(entryHostQuery).All(&hostEntries)
 
-						if res2[0].CID != a.chunk {
-							// fmt.Println("host existing", a.chunk, res2, data.Host)
-							newRecordFlag = true
-						}
+					//TODO: I think there might be a bug here. From what I can tell, if we set the new record flag
+					// when there is a rare signature record from a previous chunk, then we will end up pushing a duplicate
+					// record for each chunk. We should investigate removing this chunk check here. -LL
```

Comment:
I suppose this is somewhat the same issue I have with the max duration query. In the max duration query, we update old chunks with new measurements. Here, we always push a new entry per chunk. How does the consuming logic work for this data, and could we standardize on method or the other?